src/Core/File/CameraData.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef RADIUMENGINE_CAMERA_DATA_HPP
+#define RADIUMENGINE_CAMERA_DATA_HPP
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <Core/Containers/VectorArray.hpp>
+#include <Core/File/AssetData.hpp>
+#include <Core/Math/LinearAlgebra.hpp>
+#include <Core/RaCore.hpp>
+
+namespace Ra {
+
+namespace Asset {
+
+/// The CameraData class stores the data related to one of the Cameras of a loaded scene.
+class RA_CORE_API CameraData : public AssetData {
+
+  public:
+    EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+    /// The projection type of the Camera.
+    enum CameraType { ORTHOGRAPHIC, PERSPECTIVE };
+
+    CameraData( const std::string& name = "", const CameraType& projType = PERSPECTIVE );
+
+    CameraData( const CameraData& data ) = default;
+
+    ~CameraData();
+
+    /// Return the name of the Camera.
+    inline void setName( const std::string& name );
+
+    /// Return the projection type of the Camera.
+    inline CameraType getType() const;
+
+    /// Set the projection type of the Camera.
+    inline void setType( const CameraType& type );
+
+    /// Return the transformation of the Camera.
+    inline Core::Matrix4 getFrame() const;
+
+    /// Set the transformation of the Camera.
+    inline void setFrame( const Core::Matrix4& frame );
+
+    /// Return the Field of View of the Camera.
+    inline Scalar getFov() const;
+
+    /// Set the Field of View of the Camera.
+    inline void setFov( Scalar fov );
+
+    /// Return the distance from the near plane to the Camera.
+    inline Scalar getZNear() const;
+
+    /// Set the distance from the near plane to the Camera.
+    inline void setZNear( Scalar zNear );
+
+    /// Return the distance from the far plane to the Camera.
+    inline Scalar getZFar() const;
+
+    /// Set the distance from the far plane to the Camera.
+    inline void setZFar( Scalar zFar );
+
+    /// Return the zoom factor of the Camera.
+    inline Scalar getZoomFactor() const;
+
+    /// Set the zoom factor of the Camera.
+    inline void setZoomFactor( Scalar zoom );
+
+    /// Return the aspect ratio of the Camera.
+    inline Scalar getAspect() const;
```

Comment:
is aspect ratio an attribute of the camera or of the viewer/film ?

Reply:
It is an (unused) attribute of Camera, which is also defined in Assimp's Camera